ykcompile/src/lib.rs

```diff
@@ -677,6 +677,14 @@ impl<TT> TraceCompiler<TT> {
     }
 
     fn c_binop(&mut self, dest: &IPlace, op: BinOp, opnd1: &IPlace, opnd2: &IPlace, checked: bool) {
+        let opnd1_ty = SIR.ty(&opnd1.ty());
+        debug_assert!(opnd1_ty == SIR.ty(&opnd2.ty()));
+
+        // For now this whole function assumes we are operating on integers.
```

Comment:
Is this true? I think `c_condition` doesn't really care about the types of the operands.

Reply:
It uses `cmp` which can only be used to compare integers.

For example, to compare floats or doubles, we'd have to use one of the other comparison instructions.